servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ConnectionObserver.java

```diff
@@ -169,14 +178,18 @@ default void connectionWritabilityChanged(boolean isWritable) {
 
     /**
      * Callback when the connection is closed due to an {@link Throwable error}.
+     * <p>
+     * Connections can be closed at any time, even before they are fully established.
```

Comment:
Consider here and with the next docblock to `@link` and mention that the open callback will not have been called if this happens

Reply:
I got your point, but made changes in a slightly different way:
1. moved "terminal" clarification from the top of the interface to each "closed" method and clarified that any callback after that doesn't make sense
2. moved `connectionWritabilityChanged` up to read/write/flush events to make "established" and "closed" callbacks next to each other in the interface structure
3. added clarification that only one "established" callback can be invoked